python/ray/ml/train/integrations/lightgbm/lightgbm_trainer.py

```diff
@@ -0,0 +1,64 @@
+from typing import Optional, Dict, Any
+
+from ray.ml.trainer import GenDataset, Trainer
+from ray.ml.config import ScalingConfig, RunConfig
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+from ray.util import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+class LightGBMTrainer(Trainer):
+    """A Trainer for data parallel LightGBM training.
+
+    This Trainer runs the LightGBM training loop in a distributed manner
+    using multiple Ray Actors.
+
+    Example:
+        .. code-block:: python
+
+            import ray
+
+            from ray.ml.train.integrations.lightgbm import LightGBMTrainer
+
+            train_dataset = ray.data.from_items(
+                [{"x": x, "y": x + 1} for x in range(32)])
+            trainer = LightGBMTrainer(
+                label_column="y",
+                lightgbm_config={"objective": "regression"},
+                scaling_config={"num_workers": 3},
+                datasets={"train": train_dataset}
+            )
+            result = trainer.fit()
+
+
+    Args:
+        label_column: Name of the label column. A column with this name
+            must be present in the training dataset.
+        lightgbm_config: LightGBM training parameters. Refer to
+            `LightGBM documentation <https://lightgbm.readthedocs.io/en\
+/latest/Parameters.html>`_ for a list of possible parameters.
+        scaling_config: Configuration for how to scale data parallel training.
+        run_config: Configuration for the execution of the training run.
+        datasets: Any Ray Datasets to use for training. Must include a
+            "train" key denoting the training dataset. If a ``preprocessor``
+            is provided and has not already been fit, it will be fit on the training
+            dataset. All datasets will be transformed by the ``preprocessor`` if
+            one is provided. All non-training datasets will be used as separate
+            validation sets, each reporting a separate metric.
+        preprocessor: A ray.ml.preprocessor.Preprocessor to preprocess the
+            provided datasets.
+        resume_from_checkpoint: A checkpoint to resume training from.
+    """
+
+    def __init__(
+        self,
+        label_column: str,
+        lightgbm_config: Optional[Dict[str, Any]] = None,
+        scaling_config: Optional[ScalingConfig] = None,
+        run_config: Optional[RunConfig] = None,
+        datasets: Optional[Dict[str, GenDataset]] = None,
```

Comment:
question, should this be made a mandatory argument? the train dataset is always required.

Reply:
IMO, I think so! Like you said, train dataset is always required anyways.

Reply:
ok let's go with that